src/docfx/build/Build.cs

```diff
@@ -100,13 +102,12 @@ private static bool ShouldBuildFile(Document itemToBuild, ConcurrentDictionary<D
                 return false;
             }
 
-            if (!set.TryAdd(itemToBuild, 0))
+            if (itemToBuild.ContentType == ContentType.TableOfContents && !tocMap.Contains(itemToBuild))
```

Comment:
>!tocMap.Contains(itemToBuild) [](start = 74, length = 29)

how can this happen? should we assert?

Reply:
nested toc should not be built, i didn't change the logic here, but switch the order.

---
In reply to: [189461762](https://github.com/dotnet/docfx/pull/2787#discussion_r189461762) [](ancestors = 189461762)

Reply:
I see. looks good to me.

---
In reply to: [189479687](https://github.com/dotnet/docfx/pull/2787#discussion_r189479687) [](ancestors = 189479687,189461762)